### PR TITLE
TiFShared integration into TiFBackendUtils

### PR DESCRIPTION
**Pull request by seanim0920**
## Purpose
Start with getting tifshared to work in backendutils package before moving on to the lambdas.

## Proposed Solutions
* We had issues performing imports without file extensions in aws, forcing us to add ".js" to the end of every file path.
image
Currently TiFShared does not include those file extensions in its imports, and I didn't want to change all the files to enforce that, so I did more research and found that commonjs packages will automatically add the .js extension when resolving imports while ESM packages always require file extensions to be added to imports. This means we don't need to enforce file extensions on our imports at all and can simply build our packages in commonjs to fix the issue.
This required some small modifications like ensuring no high level awaits in our scripts and that we use __dirname instead of import.meta.

esm
image https://nodejs.org/api/esm.html#mandatory-file-extensions

commonjs
image https://nodejs.org/api/modules.html#file-modules

* In order to focus the changes on tifbackendutils for this pr and not touch the other lambdas, I re-exported the necessary modules from tifshared from tifbackendutils/index. In subsequent PR's I'll make tifshared a peer dependency rather than direct dependency so this re-exporting isn't needed and the lambdas can import from tifshared directly.

* Certain types were not being inferred properly from TiFShared in the IDE. Extending the TiFShared tsconfig helped.
image
* However, when building to js, the compiler didn't recognize the tifshared zod extensions, so I needed to import from the file directly.
image

* When importing helper types from TiFBackendUtils/TiFShared, needed to use "import type" syntax so these type imports don't appear in the js transpilation.
image

* Move util functions from TiFBackendUtils to TiFShared and fix imports
* Adjusted tsconfigs to allow TiFShared imports
* Updated typescript dependencies to match TiFShared versions

## Results
Backend is built to commonjs now
Imports replaced with TiFShared:
* LocationCoordinate2D
* Result utils (PromiseResult, Success, Failure, etc)
* NullablePartial type
* zod optionalParseable util
* UserHandle
* EventHandle
* sleep
Staging tests seem to run twice as fast, unsure if these changes caused it though

## Warnings
* Ensure you have TiFShared linked locally or the npm install won't work. Instructions can be found in the tifshared readme https://github.com/tifapp/TiFShared?tab=readme-ov-file#local-development-setup
* Avoid importing TiFBackendUtils/conn outside of the lambdas, leads to import issues

## Other Changes
* Lint timeout was too long, shortened lint timeout
* To make builds faster, prune production dependencies rather than doing another npm ci when building the lambdas
* For ease of development, added automatic TiFShared linking after local npm install. This allows local changes to tifshared to instantly be seen when developing for the backend.
* Changed module imports in GeocodingLambda and APILambda to match TiFShared (ex. UserHandle.schema -> UserHandleSchema, LocationCoordinates2D -> LocationCoordinate2D)
* Fixed LocationCoordinate2DSchema import typo (from LocationCoordinates2DSchema)
* Adjusted eslint to not require .js extensions for every import. Managed to remove some ```/* eslint-disable import/extensions */``` lines after that
* Disabled strict setting when doing npm builds since TiFShared contains some code that breaks strict rules such as https://github.com/tifapp/TiFShared/blob/13da3abbfe1f66e40670881b883dff2e8abd7f08/domain-models/User.ts#L224

# To Do
* Now that the file extensions aren't needed, in subsequent prs I'll remove the .js file extensions on all our imports.
* Make TiFShared a direct dependency of our lambdas

https://trello.com/c/xYE5fpP3/579-tifshared-package-integration-backend

### CODE REVIEW

File: TiFBackendUtils/index.ts [hunk 1]
```diff
@@ -1,12 +1,17 @@
+import "TiFShared/lib/Math.js"; // needed to compile.
```
Comment (seanim0920) [inline TiFBackendUtils/index.ts:1]: Typescript compiler did not recognize extensions from TiFShared, so I needed to manually add these imports

**Comment by tifapp** [comment]
Linked Ticket: https://trello.com/c/xYE5fpP3

File: scripts/dbToTs.ts [hunk 2]
```diff
@@ -21,77 +22,73 @@ if (process.argv.includes('--run')) {
...
+    })
   });
-  
+
   (async () => {
     try {
-      const schemaPath = path.join(__dirname, "../schema.md");
-      const entitiesPath = path.join(__dirname, "../entities.ts");
-  
+      await resetDB()
```
Comment (seanim0920) [inline scripts/dbToTs.ts:67]: need to avoid high level await for commonjs

File: scripts/dbToTs.ts [hunk 3]
```diff
@@ -21,77 +22,73 @@ if (process.argv.includes('--run')) {
       number: ["bigint"]
     },
     typeOverrides: {
-      [`${envVars.ENVIRONMENT}.TifEventView.hasArrived`]: "boolean" as const
+      [`${envVars.DATABASE_NAME}.TifEventView.hasArrived`]: "boolean" as const
```
Comment (seanim0920) [inline scripts/dbToTs.ts:25]: ENVIRONMENT was invalid, should've been using DATABASE_NAME after this pr https://github.com/tifapp/FitnessProjectBackend/pull/243/files

**Review by mhayes853 — APPROVED** [review]

**Pull request merged**